src/visit/mod.rs

```diff
@@ -398,6 +398,13 @@ pub trait VisitMap<N> {
 
     /// Return whether `a` has been visited before.
     fn is_visited(&self, a: &N) -> bool;
+
+    /// Mark `a` as unvisited.
+    ///
+    /// Return **true** if this vertex was marked as visited at the time of unsetting it, false otherwise.
+    fn unvisit(&mut self, _a: N) -> bool {
+        unimplemented!("We don't know how to mark the node as unvisited.")
```

Comment:
Could it be avoided to have a default implementation? Instead leave this out and let it be a breaking change, that's the principled approach at least.

Reply:
Yeah we could totally get by without a default implementation. My initial idea was to not make this a breaking change but an additional feature that is backwards compatible. I don't have a strong preference but if breaking change is preferable, then lets go ahead with that.